src/travel-details-screens/utils.ts

```diff
@@ -321,138 +322,93 @@ export function getSecondsRemainingToBookingDeadline(
-  bookingArrangements: BookingArrangementFragment,
-  aimedStartTime: string,
-  now: number,
-  flex_booking_number_of_days_available: number,
-): boolean {
-  const secondsInNumberOfDaysAvailable =
-    flex_booking_number_of_days_available * 24 * secondsInOneHour;
-  const secondsRemainingToDeadline = getSecondsRemainingToBookingDeadline(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-  );
-  return (
-    secondsRemainingToDeadline >= secondsInOneHour &&
-    secondsRemainingToDeadline > secondsInNumberOfDaysAvailable
-  );
-}
-
-export function getIsTooLateToBook(
-  bookingArrangements: BookingArrangementFragment,
-  aimedStartTime: string,
-  now: number,
-): boolean {
-  const secondsRemainingToDeadline = getSecondsRemainingToBookingDeadline(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-  );
-  return secondsRemainingToDeadline < 0;
-}
-
-export function getBookingIsAvailableImminently(
-  bookingArrangements: BookingArrangementFragment,
-  aimedStartTime: string,
-  now: number,
-  flex_booking_number_of_days_available: number,
-): boolean {
-  const secondsRemainingToAvailable = getSecondsRemainingUntilBookingAvailable(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-    flex_booking_number_of_days_available,
-  );
-  return (
-    secondsRemainingToAvailable > 0 &&
-    secondsRemainingToAvailable < secondsInOneHour
-  );
-}
-
-export function getIsBookingAvailable(
+/**
+ * Get bookings status based on the given booking arrangements and aimed start
+ * time. Will return one of these values:
+ * - 'none': When there is no booking arrangements
+ * - 'early': It is too early to book the trip
+ * - 'bookable': The trip may be booked
+ * - 'late': The booking deadline has passed
+ * '´
+ * @param bookingArrangements The booking arrangements for the leg or estimated
+ * call
+ * @param aimedStartTime the aimed start time of the leg or estimated call
+ * @param now The now value to use, if not given the Date.now() is used
+ * @param flex_booking_number_of_days_available Number of days before latest
+ * booking time the user may book, if not given then the returned status can
+ * never be 'early'.
+ */
+export const getBookingStatus = (
   bookingArrangements: BookingArrangementFragment | undefined,
   aimedStartTime: string,
-  now: number,
-  flex_booking_number_of_days_available: number,
-): boolean {
-  if (!bookingArrangements) return false;
+  now: number = Date.now(),
+  flex_booking_number_of_days_available?: number,
+): BookingStatus => {
+  if (!bookingArrangements) return 'none';
 
-  const isTooEarly = getIsTooEarlyToBook(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-    flex_booking_number_of_days_available,
-  );
-  const isTooLate = getIsTooLateToBook(
+  const secondsToDeadline = getSecondsRemainingToBookingDeadline(
     bookingArrangements,
     aimedStartTime,
     now,
   );
 
-  return !isTooEarly && !isTooLate;
-}
+  if (secondsToDeadline < 0) {
```

Comment:
should this be based on the booking max time?, as far I remember we discussed it to be 3h before departure, but this is not a rule written somewhere because this value should come in the response. It might be OK to have it like now but would be better if is a real time.

Reply:
`aimedStartTime` seems to be the departure time. Not the time when it is OK to book it.

Reply:
When you say booking max time, do you mean the booking deadline? If so that is calculated in the `getSecondsRemainingToBookingDeadline` based on the data returned from Entur. For example for the Vy bus shown in the pictures the deadline is 15 minutes before the start time. And if seconds to booking deadline is negative, then it is too late to book it.

Reply:
OK, that sounds good, I just saw `aimedStart` in the equation. seems like latestBookingTime inside booking arrangement.